docs/developer/running-saleor/read-replicas.mdx

```diff
@@ -0,0 +1,85 @@
+---
+title: Database read replica
+---
+
+Database read replica is a read-only copy of the primary database. Replica should reflect changes in the primary database as fast as possible. This feature allows Saleor to offload the primary database by moving read-only
+traffic to a separate database.
+:::note
+Asynchronous replication is not part of Saleor Core. It's should be configured separately depending on the environment.
+:::
+```mermaid
+graph LR
+  C((Saleor Core)) ---|Read/Write| P[(Primary database)]
+  C ---|Read| R[(Read replica)]
+  P --->|Asynchronous replication| R
+```
+
+## How to configure
+Django allows using [multiple databases.](https://docs.djangoproject.com/en/4.0/topics/db/multi-db/)
+To start using read replica in Saleor we should set `DATABASE_CONNECTION_REPLICA_NAME` in `settings.py` and specify this connection in `DATABASES`.
+
+### Example config:
+:::note
+Primary database: postgres://saleor:saleor@localhost:5432/saleor
+
+Read replica: postgres://saleor:saleor@localhost:5433/saleor
+:::
+
+In this case, we should set `DATABASE_CONNECTION_REPLICA_NAME` to `replica`.
+
+In `DATABASES` we should add new entry:
+`DATABASE_CONNECTION_REPLICA_NAME: dj_database_url.config(default="postgres://saleor:saleor@localhost:5433/saleor",conn_max_age=600)`,
+
+
+## How to use replica in code
+
+By default Saleor code always use the primary database.
+
+To specify which connection should be used by ORM we use [`using` on queryset.](https://docs.djangoproject.com/en/4.0/topics/db/multi-db/#manually-selecting-a-database)
+
+> **Example**
+>
+> To fetch all products from the primary database we could use:
+>
+> - Products.objects.all()
+> - Products.objects.using(DATABASE_CONNECTION_DEFAULT_NAME).all()
+>
+> To fetch all products from the read replica we could use:
+>
+> - Products.objects.using(DATABASE_CONNECTION_REPLICA_NAME).all()
+
+:::note
+Do not forget about importing the proper connection name from settings.
+:::
+
+
+### How to use read replicas in dataloaders
+
+Dataloaders could be used to make responses for mutations and queries. Mutations should always return data from the primary database to show real-time data as a result of mutation. To handle this situation we introduce
+`database_connection_name` as a part of the base `Dataloader` class. All dataloaders ORM queries should use this variable as connection name
+> **Example**
+>
+>
+> Product.objects.using(self.database_connection_name).in_bulk(keys)
+
+
+## How to test it in a local machine
+
+To simulate read replica on the local machine we could use the second connection to the primary database with the read-only user.
+
+To setup it we should add read-only user to database
```

Comment:
1. Add a read-only user in your database:
```
SQL example to create the user
```
If you're using the [saleor-platform](link), you will have a read-only user created automatically.

2. Add the following configuration in `settings.py`:

```
full example from settings.py
```